lib/device-log/safari-network-log.js

```diff
@@ -166,6 +173,28 @@ class SafariNetworkLog extends RotatingLog {
       this.log.debug(`  Cancelled: ${cancelled}`);
     }
   }
+
+  async getLogs () {
+    if (this.logs.length && this.logIdxSinceLastRequest < this.logs.length) {
+      let result = _.clone(this.logs);
+      if (this.logIdxSinceLastRequest > 0) {
+        result = result.slice(this.logIdxSinceLastRequest);
```

Comment:
I assume slice anyway performs cloning, so we are spending extra time there to do double cloning

Reply:
Slice does a shallow copy. So this is still needed since it is a nested structure. Though only Appium uses the result, so the whole clone is not needed if we can assume we're not going to do something stupid in the calling code.